lib/db.js

```diff
@@ -84,7 +93,62 @@ class LimitDBRedis extends EventEmitter {
     });
   }
 
+  #setupPing() {
+    if (!this.pingConfig?.enabled) {
+      return;
+    }
+
+    this.failedPings = 0;
+    this.once('ready', () => {
+      this.#startPing();
+    })
+  }
+
+  #startPing() {
+    this.keepPingRunning = true;
+    const doPing = () => {
+      let hrstart = process.hrtime();
+      this.redis.ping((err, payload) => {
+        let hrend = process.hrtime(hrstart);
```

Comment:
Thinking if we emit when we start pinging we can remove the duration computation and do it in auth0-limitd-redis. But not sure which is the best solution.

Reply:
Yes! ioredis already have a way of set a command timeout; maybe we can use that as a strategy? https://github.com/luin/ioredis/blob/ea8a006daabb7ade0a2cedcaade6543766a351e7/lib/redis/RedisOptions.ts#LL16C3-L16C17